src/firebolt/model/V2/engine.py

```diff
@@ -203,21 +208,31 @@ def update(
             )
 
         sql = self.ALTER_PREFIX_SQL.format(self.name)
-        parameters = []
-        for param, value in zip(
-            self.ALTER_PARAMETER_NAMES,
-            (scale, spec, auto_stop, name),
-        ):
-            if value is not None:
-                sql_part, new_params = self._service._format_engine_attribute_sql(
-                    param, value
+        parameters: List[Union[str, int]] = []
+        if name is not None:
+            if not self._engine_name_re.match(name):
```

Comment:
I'm thinking from a readability perspective, we can potentially do `name.replace('_', '').isalnum()` here and avoid regex altogether.
Don't feel too strongly about it, so up to you.

Reply:
I would rather stick to the regexp as it fells more straightforward, while removing characters might be slightly confusing during reading